Lib/fontTools/ttLib/tables/_n_a_m_e.py

```diff
@@ -223,6 +227,28 @@ def removeNames(self, nameID=None, platformID=None, platEncID=None, langID=None)
             )
         ]
 
+    @staticmethod
+    def removeUnusedNames(ttFont):
+        """Remove any name records which are not in NameID range 0-255 and not utilized
+        within the font itself."""
+        visitor = NameRecordVisitor()
+        visitor.visit(ttFont)
+        toDelete = set()
+        for record in ttFont["name"].names:
+            # Name IDs 26 to 255, inclusive, are reserved for future standard names.
+            # https://learn.microsoft.com/en-us/typography/opentype/spec/name#name-ids
+            if record.nameID < 256:
+                continue
+            if record.nameID not in visitor.seen:
+                toDelete.add(record.nameID)
+
+        if not toDelete:
+            log.info("Name table has no redundant records, skipping")
```

Comment:
demote to DEBUG or remove entirely? Maybe the method could return the set of nameIDs that it has deleted so that caller can decide whether they wish to log this or not

Reply:
I've removed the log. Method now returns the deleted records nameIDs.